docs/tutorials/python/sagas/index.md

```diff
@@ -0,0 +1,586 @@
+    worker = Worker(
+        client,
+        task_queue=TASK_QUEUE_NAME,
+        workflows=[BookWorkflow],
+        activities=[
+            book_car,
+            book_hotel,
+            book_flight,
+            undo_book_car,
+            undo_book_hotel,
+            undo_book_flight,
+        ],
+    )
+    await worker.run()
+    try:
+        await interrupt_event.wait()
+    finally:
+        print("\nShutting down the worker\n")
+
+
+if __name__ == "__main__":
+    loop = asyncio.get_event_loop()
+    try:
+        loop.run_until_complete(main())
+    except KeyboardInterrupt:
+        print("\nInterrupt received, shutting down...\n")
+        interrupt_event.set()
+        loop.run_until_complete(loop.shutdown_asyncgens())
+```
+
+The `Client.connect()` line connects to the Temporal server running on `localhost` at port `7233`.
+This can be modified to run a Worker on Temporal Cloud.
+
+The `Worker` is initialized with the client, the Task Queue name, the list of Workflows, and the list of Activities.
+This setup ensures the Worker knows which tasks to listen for and execute.
+The `await worker.run()` line starts the Worker, making it ready to receive tasks and execute the corresponding Activities and Workflows.
+
+To start the Worker, run the following command in your terminal:
+
+```bash
+# terminal one
+python run_worker.py
+```
+
+Once the Worker is running, it will be ready to execute Workflows and Activities as tasks are submitted to the specified Task Queue.
+
+Now that the Worker is set up and running, you can set up the Client to initiate the booking process.
+
+## Build the Flask API to book trips
+
+To make your booking logic useful, you need to integrate it into an application.
+You will create a Flask endpoint that accepts posts requests, so you can send data to the Workflow to be processed by the Activities.
+
+This setup allows you to interact with the Temporal service and trigger the booking Workflow through HTTP requests.
+
+Create a new file named `run_workflow.py`.
+
+Import the necessary modules, including `uuid`, Flask, and Temporal `Client`.
+
+```python
+import uuid
+
+from flask import Flask, jsonify, request
+from temporalio.client import Client
+
+from activities import BookVacationInput
+from saga_workflows import BookWorkflow
+from shared import TASK_QUEUE_NAME
+```
+
+The `uuid` module is used to generate a unique ID for each booking.
+The `Flask` module is used to set up the Flask API.
+The `Client` module is used to connect to the Temporal Service.
+
+Initialize the Flask app and set up the dependency injection for the Temporal Client.
+
+```python
+app = Flask(__name__)
+
+
+async def get_temporal_client():
+    return await Client.connect("localhost:7233")
+```
+
+Dependency injection for the Temporal Client is used here to ensure it is initialized once and reused, avoiding the resource-intensive process of repeatedly starting it for each booking request.
```

Comment:
> Dependency injection for the Temporal Client is used here

Where is it used here? The code in this README calls `get_temporal_client()` directly from the HTTP  handler